test/compilable/isZeroInit.d

```diff
@@ -0,0 +1,50 @@
+alias AliasSeq(T...) = T;
+
+struct Holder(T, T val)
+{
+    T x = T(val);
+}
+
+static foreach (T; AliasSeq!(bool, byte, short, int, long,
+                             ubyte, ushort, uint, ulong,
+                             char, wchar, dchar,
+                             float, double, real))
+{
+    static assert(__traits(isZeroInit, T) == (T.init is T(0)));
+    static assert(__traits(isZeroInit, T[2]) == (T.init is T(0)));
+
+    static assert(!__traits(isZeroInit, Holder!(T, 1)));
+    static assert(__traits(isZeroInit, Holder!(T, 0)));
+}
+
+static assert(__traits(isZeroInit, void)); // For initializing arrays of element type `void`.
+static assert(__traits(isZeroInit, void*));
```

Comment:
Could we add a test for `static assert(__traits(isZeroInit, void[]));` too?

Reply:
Do you mean `void[10]`? Every dynamic array has a zero initializer.